src/Provider/OfflineRefundPaymentMethodsProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Provider;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Repository\PaymentMethodRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class OfflineRefundPaymentMethodsProvider implements RefundPaymentMethodsProviderInterface
+{
+    /** @var PaymentMethodRepositoryInterface */
+    private $paymentMethodRepository;
+
+    public function __construct(PaymentMethodRepositoryInterface $paymentMethodRepository)
+    {
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    public function findForChannel(ChannelInterface $channel): array
+    {
+        $paymentMethods = $this->paymentMethodRepository->findEnabledForChannel($channel);
+
+        /** @var PaymentMethodInterface $paymentMethod */
+        foreach ($paymentMethods as $key => $paymentMethod) {
+            Assert::notNull($paymentMethod->getGatewayConfig());
+            if ($paymentMethod->getGatewayConfig()->getFactoryName() !== 'offline') {
+                unset($paymentMethods[$key]);
+            }
+        }
+
+        return $paymentMethods;
```

Comment:
Not a big one, but you can get rid of that `unset` and phpdoc by using `array_filter`:

```php
return array_filter(
  $this->paymentMethodRepository->findEnabledForChannel($channel),
  function (PaymentMethodInterface $paymentMethod): bool {
    Assert::notNull($paymentMethod->getGatewayConfig());

    return $paymentMethod->getGatewayConfig()->getFactoryName() === 'offline';
  }
);
```

Reply:
Good idea :